### ISSUE TITLE
PySCF produces wrong Molden input file

### ISSUE THREAD

**Issue by jonas-greiner** [pyscf/pyscf#2287]
Hi everyone,

I have found an issue with the Molden file produced by PySCF. For the following minimal example, molden is unable to visualize the orbitals:

```
from pyscf import gto, scf, tools

mol = gto.Mole()
mol.build(
    atom="""
        C
        H  1  1.1015
        H  1  1.1015  2  109.5
        H  1  1.1015  2  109.5  3  120.0
        H  1  1.1015  2  109.5  3  240.0
    """,
    basis="cc-pVDZ",
)

hf = scf.RHF(mol).run()

tools.molden.from_mo(mol, "mo_coeff.molden", hf.mo_coeff)
```

This also appears to be the case for some other symmetric molecules. I was able to fix this issue by shifting the coordinates in the header to the center of mass and will add a corresponding PR.

**Comment by sunqm** [comment]
Which molden visualizer do you use? I can view the orbitals using Jmol without any problems

**Comment by jonas-greiner** [comment]
I used regular molden (version 6.9) from [https://ftp.science.ru.nl/Molden/](https://ftp.science.ru.nl/Molden/) and figured this out by comparing with molden files generated by other programs. I actually just tested it with a more recent version of molden (version 7.3) and it appears to work without shifting the coordinates. If you want it to work with older versions it could be worth making the changes in the PR, otherwise I can cancel it.
